### ISSUE TITLE
liquid_ssb.c

### ISSUE THREAD

**Issue by Claudio-Sjo** [F5OEO/rpitx#336]
the file liquid_ssb.c is missing for generating pissb

**Comment by mfalkvidd** [comment]
Sorry for not checking myself (I don’t have access to a linux computer at the moment) but see if `sudo apt-get install libliquid-dev` helps.

edit: sorry, that won’t help. Seems like the file has been missing since the initial commit 6 years ago.

**Comment by Claudio-Sjo** [comment]
Hi,
actually it doesn't.
The file is supposed to exist in sssbgen directory

This is from the Makefile in src directory

../pissb: ssbgen/test_ssb.c ssbgen/ssb_gen.c ssbgen/liquid_ssb.c
        $(CC) $(CFLAGS_Pissb) -o ../pissb ssbgen/liquid_ssb.c $(LDFLAGS_Pissb)

**Comment by barf** [comment]
Check your GCC options, it builds here but the detect script found wrong Pi version

**Comment by ticklemynausea** [comment]
I'm also stumped trying to build this, don't really know where to turn to. No references in the enitre git log to this `liquid_ssb.c` file :/ wtf

**Comment by ticklemynausea** [comment]
well I managed to build `pissb` with the following changes to `src/Makefile`:

```
diff --git a/src/Makefile b/src/Makefile
index 9ea8b70..f838882 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -11,8 +11,8 @@ INSTALL_DIR ?= /usr/local/bin
 CFLAGS_Pissb   = -Wall -g -O2 -Wno-unused-variable
 LDFLAGS_Pissb  = $(LDFLAGS) -lsndfile -lliquid

-../pissb: ssbgen/test_ssb.c ssbgen/ssb_gen.c ssbgen/liquid_ssb.c
-       $(CC) $(CFLAGS_Pissb) -o ../pissb ssbgen/liquid_ssb.c $(LDFLAGS_Pissb)
+../pissb: ssbgen/test_ssb.c ssbgen/ssb_gen.c
+       $(CC) $(CFLAGS_Pissb) -o ../pissb ssbgen/test_ssb.c ssbgen/ssb_gen.c $(LDFLAGS_Pissb)

 ../pisstv : sstv/pisstv.cpp
        $(CXX) $(CXXFLAGS) -o ../pisstv sstv/pisstv.cpp  $(LDFLAGS)
```

**Comment by barf** [comment]
Thanks @ticklemynausea this patch worked but I'm not sure this is needed for SSB. USB can be sent using `sendiq` and some csdr pipes, see the example `testssb.sh`. Also some examples required linking with the C math library, so I added `-lm` to the LDFLAGS too.

I can't find the file either but it only prevented building this example, the other SSB example works fine.

To send lower sideband just invert the I and Q (swap the interleaving).

**Comment by ticklemynausea** [comment]
Yes probably the broken parts of the Makefile aren't really needed for the examples. I still have a lot of exploring left to do.
